sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -895,14 +895,90 @@ def test_non_euclidean_kneighbors():
     # Raise error when wrong parameters are supplied,
     X_nbrs = neighbors.NearestNeighbors(3, metric='manhattan')
     X_nbrs.fit(X)
-    assert_raises(ValueError, neighbors.kneighbors_graph, X_nbrs, 3,
+    assert_raises(ValueError, neighbors.kneighbors_graph, None, 3,
                   metric='euclidean')
     X_nbrs = neighbors.NearestNeighbors(radius=radius, metric='manhattan')
     X_nbrs.fit(X)
-    assert_raises(ValueError, neighbors.radius_neighbors_graph, X_nbrs,
+    assert_raises(ValueError, neighbors.radius_neighbors_graph, None,
                   radius, metric='euclidean')
 
 
+def test_kneighbors_graph_nearest_neighbors():
+    """Test kneighbors_graph method in NearestNeighbors"""
+    for algorithm in ["kd_tree", "ball_tree", "brute"]:
+
+        nn = neighbors.NearestNeighbors(n_neighbors=1, algorithm=algorithm)
+
+        X = [[0], [1]]
+        nn.fit(X)
+
+        # Do not do anything special to duplicates.
+        test_data = [[0], [1]]
+        assert_array_equal(
+            nn.kneighbors_graph(test_data).A, [[1., 0.], [0., 1.]])
+        dist, ind = nn.kneighbors([[0], [1]])
+        assert_array_equal(dist, [[0], [0]])
+        assert_array_equal(ind, [[0], [1]])
+
+        test_data = [[2], [1]]
+        assert_array_equal(
+            nn.kneighbors_graph(test_data).A, [[0., 1.], [0., 1.]])
+        dist, ind = nn.kneighbors(test_data)
+        assert_array_equal(dist, [[1], [0]])
+        assert_array_equal(ind, [[1], [1]])
+
+        # Avoid marking sample to itself when X is None
+        assert_array_equal(
+            nn.kneighbors_graph().A,
+            np.array([[0., 1.], [1., 0.]]))
+        dist, ind = nn.kneighbors()
+        assert_array_equal(dist, [[1], [1]])
+        assert_array_equal(ind, [[1], [0]])
+
+        # Do not do anything special to duplicates.
+        kng = nn.kneighbors_graph([[0], [1]], mode='distance')
+        assert_array_equal(
+            kng.A,
+            np.array([[0., 0.], [0., 0.]]))
```

Comment:
Not to self and other reviewers: this is a non-regression test that wouldn't pass on master...